### PR TITLE
[SPARK-6095] [MLLIB] Support model save/load in Python's linear models

### PR DESCRIPTION
**Pull request by yanboliang**
For Python's linear models, weights and intercept are stored in Python.
This PR implements Python's linear models sava/load functions which do the same thing as scala. 
It can also make model import/export cross languages.

### CODE REVIEW

**Comment by yanboliang** [comment]
retest, please.

**Comment by mengxr** [comment]
@yanboliang Sorry for miscommunication! My original suggestion was to reuse the Scala implementation to achieve cross-language data compatibility but not have a separate implementation in Python. Please check the changes in https://github.com/apache/spark/pull/5049 and see whether we can explore the same structure for linear models. Thanks!

File: python/pyspark/mllib/classification.py [hunk 1]
```diff
@@ -124,6 +136,22 @@ def predict(self, x):
         else:
             return 1 if prob > self._threshold else 0
 
+    def save(self, sc, path):
+        java_model = sc._jvm.org.apache.spark.mllib.classification.LogisticRegressionModel(
+            _py2java(sc, self._coeff), _py2java(sc, self.intercept))
```
Comment (mengxr) [inline python/pyspark/mllib/classification.py]: `intercept` is a float, we don't need `_py2java` for it.

File: python/pyspark/mllib/regression.py [hunk 2]
```diff
@@ -54,6 +54,10 @@ def __repr__(self):
         return "LabeledPoint(%s, %s)" % (self.label, self.features)
 
 
+from pyspark.mllib.util import Saveable
+from pyspark.mllib.util import Loader
```
Comment (mengxr) [inline python/pyspark/mllib/regression.py]: Should it be `from pyspark.mllib.util import Saveable, Loader` and stay at the top of the file?

**Comment by mengxr** [comment]
LGTM except minor inline comments.

**Comment by yanboliang** [comment]
@mengxr Thank you for your comments.

**Comment by mengxr** [comment]
LGTM. Merged into master. Thanks!!

**Pull request closed without merging**